spec/awful/completion_spec.lua

```diff
@@ -55,8 +60,20 @@ describe("awful.completion.shell", function()
     end)
 
     it("completes local file", function()
-        assert.same(shell('ls ', 4, 1, 'zsh'), {'ls localcommand', 16, {'localcommand'}})
-        assert.same(shell('ls ', 4, 1, 'bash'), {'ls localcommand', 16, {'localcommand'}})
+        assert.same(shell('ls l', 5, 1, 'zsh'), {'ls localcommand', 16, {'localcommand'}})
+        assert.same(shell('ls l', 5, 1, 'bash'), {'ls localcommand', 16, {'localcommand'}})
+    end)
+
+    it("completes command together with local directory", function()
+        assert.same(shell('true', 5, 1, 'zsh'), {'true', 5, {'true', 'true/'}})
+        assert.same(shell('true', 5, 1, 'bash'), {'true', 5, {'true', 'true/'}})
+        assert.same(shell('true', 5, 1, nil), {'true', 5, {'true', 'true/'}})
+    end)
+
+    it("completes local directory", function()
+        -- TODO: Does not work with zsh yet!
+        --assert.same(shell('just_a', 7, 1, 'zsh'), {'just_a_directory/', 18, {'just_a_directory/'}})
+        assert.same(shell('just_a', 7, 1, 'bash'), {'just_a_directory/', 18, {'just_a_directory/'}})
```

Comment:
I can look into Zsh here, but think that it should be handled by a proper Zsh completion daemon (https://github.com/awesomeWM/awesome/issues/1902) in the long run.
This would then just use the user's config for completion(s).